sdk/client/client.ts

```diff
@@ -196,91 +248,83 @@ export class Client {
   The function below is a wrapper function which wraps the input function
   'apiFunction' with token refresh logic to ensure uninterrupted access to LUSID.
   */
-  public apiFunctionWrapper(apiFunction, api, self) {
-
-    // Return a function
-    return function() {
-      // Collect the arguments passed into the wrapper to use later
-      var topLevelArguments = arguments
-      // Return a promise to ensure that the function remains 'thenable'
-      return new Promise(function(resolve, reject) {
+  private apiFunctionWrapper(apiFunction, api ) {
+
+    // Return a function, thus not immediately invoking
+
+    return ( ...args ) => {
+
+      // Return a promise to ensure that the function remains '.then()-able'
+      return new Promise( (resolve, reject) => {
```

Comment:
Is "this" inside the promise referring to the Client class now? If so, how did you achieve that?